application/src/main/kotlin/application/ExamplesCommand.kt

```diff
@@ -198,81 +213,165 @@ For example:
-                            ScenarioStub(it.first, it.second)
-                        }
-                    }
+        private fun validateImplicitExamplesFrom(contractFile: File): Int {
+            val feature = parseContractFileToFeature(contractFile)
 
-                    ExamplesInteractiveServer.validateMultipleExamples(feature, examples = inlineExamples, inline = true, scenarioFilter = scenarioFilter)
-                } else emptyMap()
+            val (validateInline, validateExternal) = getValidateInlineAndValidateExternalFlags()
 
-                val externalExampleValidationResults = if(validateExternal) {
-                    val (externalExampleDir, externalExamples) = loadExternalExamples(contractFile)
+            val inlineExampleValidationResults = if (!validateInline) emptyMap()
+            else validateImplicitInlineExamples(feature)
 
-                    if(!externalExampleDir.exists()) {
-                        logger.log("$externalExampleDir does not exist, did not find any files to validate")
-                        return 1
-                    }
+            val externalExampleValidationResults = if (!validateExternal) emptyMap()
+            else {
+                val (exitCode, validationResults)
+                        = validateExamplesDir(contractFile, implicitExternalExampleDirFrom(contractFile))
+                if(exitCode == 1) exitProcess(1)
+                validationResults
+            }
 
-                    if(externalExamples.none()) {
-                        logger.log("No example files found in $externalExampleDir")
-                        return 1
-                    }
+            val hasFailures =
+                inlineExampleValidationResults.containsFailure() || externalExampleValidationResults.containsFailure()
 
-                    ExamplesInteractiveServer.validateMultipleExamples(feature, examples = externalExamples, scenarioFilter = scenarioFilter)
-                } else emptyMap()
+            printValidationResult(inlineExampleValidationResults, "Inline example")
+            printValidationResult(externalExampleValidationResults, "Example file")
 
-                val hasFailures = inlineExampleValidationResults.any { it.value is Result.Failure } || externalExampleValidationResults.any { it.value is Result.Failure }
+            if (hasFailures) return 1
+            return 0
+        }
 
-                printValidationResult(inlineExampleValidationResults, "Inline example")
-                printValidationResult(externalExampleValidationResults, "Example file")
+        private fun validateImplicitInlineExamples(feature: Feature): Map<String, Result> {
+            return ExamplesInteractiveServer.validateMultipleExamples(
+                feature,
+                examples = feature.stubsFromExamples.mapValues {
+                    it.value.map { ScenarioStub(it.first, it.second) }
+                },
+                inline = true,
```

Comment:
see if the name of the function and the param can be fixed.

Reply:
Updated the function name to - validateExamples 
But did not change the parameter name `inline` since it is not clear what purpose it is serving.